### ISSUE TITLE
multiple expectations on the same statement

### ISSUE THREAD

**Issue by ddanurwenda** [go-rel/reltest#20]
I see that `go-rel/reltest` is not merely relocated lines of code from `go-rel/rel/reltest`. One thing I still find it confusing is the feature I mentioned on the title is no longer supported in `go-rel/reltest`. For example, test below (slightly modified from `repository_test.go`) will pass on `go-rel/rel/reltest` but fail on `go-rel/reltest`

```
func TestRepository_Transaction_error(t *testing.T) {
	var (
		repo   = New()
		result = Book{Title: "Golang for dummies"}
		book   = Book{ID: 1, Title: "Golang for dummies"}
	)

	repo.ExpectTransaction(func(repo *Repository) {
		repo.ExpectInsert()
		repo.ExpectUpdate()
		repo.ExpectUpdate().ConnectionClosed()
	})

	assert.Equal(t, sql.ErrConnDone, repo.Transaction(context.TODO(), func(ctx context.Context) error {
		if err := repo.Insert(ctx, &result); err != nil {
			return err
		}
		if err := repo.Update(ctx, &result); err != nil {
			return err
		}
		return repo.Update(ctx, &result)
	}))

	assert.Equal(t, book, result)
	repo.AssertExpectations(t)
}
```

Is there any suggestion how to refactor above test to pass on `go-rel/reltest` as well? Thank you

**Comment by Fs02** [comment]
looks like the missing feature from `testify/mock` is the ability to find a better mocks.
https://github.com/stretchr/testify/blob/acba37e5db06f0093b465a7d47822bf13644b66c/mock/mock.go#L321-L323

we should try to prioritize mocks based on:
- number of matched attributes (full match or partial match using any)
- type or table match
- repeatability and total calls (repeatability > 0 or totallCalls == 0 should be prioritized)

the last criteria should makes your test pass

**Comment by Fs02** [comment]
But looks like that implementation is too complex to be added here, too many possible mocks variations (struct, partial struct, table, type, query etc) in `reltest` matcher 🤔 

My other idea is to make `repo.ExpectUpdate().Success().Times(1)` as default behavior, unless Many (`repo.ExpectUpdate().Success().Many()` is specified).

let me know if you have any thought

**Comment by ddanurwenda** [comment]
I always thought `repo.ExpectX()` _without additional assertion_ is equal to `repo.ExpectX().Success().Times(1)`, at least that's how I used `go-rel/rel/reltest`. I think setting it as default behavior is a good idea :+1:

**Comment by Fs02** [comment]
Agree,  let’s do that!

**Comment by Fs02** [comment]
this changes has been released as v0.4.0: https://github.com/go-rel/reltest/releases/tag/v0.4.0
sorry for the late release 🙏
